// @aws-sdk/middleware-location-constraint/dist/es/index.js
import { __assign, __awaiter, __generator } from "tslib";
/**
 * This middleware modifies the input on S3 CreateBucket requests.  If the LocationConstraint has not been set, this
 * middleware will set a LocationConstraint to match the configured region.  The CreateBucketConfiguration will be
 * removed entirely on requests to the us-east-1 region.
 */
export function locationConstraintMiddleware(options) {
    var _this = this;
    return function (next) {
        return function (args) { return __awaiter(_this, void 0, void 0, function () {
            var CreateBucketConfiguration, region;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        CreateBucketConfiguration = args.input.CreateBucketConfiguration;
                        return [4 /*yield*/, options.region()];
                    case 1:
                        region = _a.sent();
                        if (!CreateBucketConfiguration || !CreateBucketConfiguration.LocationConstraint) {
                            args = __assign(__assign({}, args), { input: __assign(__assign({}, args.input), { CreateBucketConfiguration: region === "us-east-1" ? undefined : { LocationConstraint: region } }) });
                        }
                        return [2 /*return*/, next(args)];
                }
            });
        }); };
    };
}
export var locationConstraintMiddlewareOptions = {
    step: "initialize",
    tags: ["LOCATION_CONSTRAINT", "CREATE_BUCKET_CONFIGURATION"],
    name: "locationConstraintMiddleware",
    override: true,
};
export var getLocationConstraintPlugin = function (config) { return ({
    applyToStack: function (clientStack) {
        clientStack.add(locationConstraintMiddleware(config), locationConstraintMiddlewareOptions);
    },
}); };
