// @aws-sdk/client-auditmanager/dist/es/commands/UpdateControlCommand.js
import { __extends } from "tslib";
import { UpdateControlRequest, UpdateControlResponse } from "../models/models_0";
import { deserializeAws_restJson1UpdateControlCommand, serializeAws_restJson1UpdateControlCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>
 * Updates a custom control in AWS Audit Manager.
 * </p>
 * @example
 * Use a bare-bones client and the command you need to make an API call.
 * ```javascript
 * import { AuditManagerClient, UpdateControlCommand } from "@aws-sdk/client-auditmanager"; // ES Modules import
 * // const { AuditManagerClient, UpdateControlCommand } = require("@aws-sdk/client-auditmanager"); // CommonJS import
 * const client = new AuditManagerClient(config);
 * const command = new UpdateControlCommand(input);
 * const response = await client.send(command);
 * ```
 *
 * @see {@link UpdateControlCommandInput} for command's `input` shape.
 * @see {@link UpdateControlCommandOutput} for command's `response` shape.
 * @see {@link AuditManagerClientResolvedConfig | config} for command's `input` shape.
 *
 */
var UpdateControlCommand = /** @class */ (function (_super) {
    __extends(UpdateControlCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function UpdateControlCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    UpdateControlCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AuditManagerClient";
        var commandName = "UpdateControlCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateControlRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateControlResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateControlCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateControlCommand(input, context);
    };
    UpdateControlCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateControlCommand(output, context);
    };
    return UpdateControlCommand;
}($Command));
export { UpdateControlCommand };
